src/System.Private.CoreLib/shared/System/Convert.cs

```diff
@@ -2518,21 +2520,131 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
                 return true;
             }
+        }      
+
+        internal static readonly Vector128<byte> s_base64ShuffleMask = Vector128.Create((byte)
+            1, 0, 2, 1, 4, 3, 5, 4, 7, 6, 8, 7, 10, 9, 11, 10);
+
+        internal static readonly Vector128<byte> s_base64ShiftLut = Vector128.Create(
+            (sbyte)'a' - 26, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'+' - 62,
+            (sbyte)'/' - 63, (sbyte)'A', 0, 0).AsByte();
+
+        internal static readonly Vector128<byte> s_base64TwoBytesStringMaskLo = Vector128.Create(
+                0, 0x80, 1, 0x80,
+                2, 0x80, 3, 0x80,
+                4, 0x80, 5, 0x80,
+                6, 0x80, 7, 0x80);
+
+        private static unsafe (int i, int j, int charcount) ConvertToBase64ArraySsse3(char* outChars, byte* inData, int length, int offset, bool insertLineBreaks)
+        {
+            int i = offset, j = 0, charcount = 0;
+            const int stride = 4 * 3;
+
+            // Based on "Base64 encoding with SIMD instructions" article by Wojciech Muła	
+            // http://0x80.pl/notesen/2016-01-12-sse-base64-encoding.html
+            byte* outputBytes = (byte*)outChars;
+
+            Vector128<byte>   tt0 = Vector128.Create(0x0fc0fc00).AsByte();
+            Vector128<ushort> tt1 = Vector128.Create(0x04000040).AsUInt16();
+            Vector128<byte>   tt2 = Vector128.Create(0x003f03f0).AsByte();
+            Vector128<ushort> tt3 = Vector128.Create(0x01000010).AsUInt16();
+            Vector128<byte>   tt5 = Vector128.Create((byte)51);
+            Vector128<sbyte>  tt7 = Vector128.Create((sbyte)26);
+            Vector128<byte>   tt8 = Vector128.Create((byte)13);
+
+            Vector128<byte> localShiftLut = s_base64ShiftLut;
+            Vector128<byte> localShuffleMask = s_base64ShuffleMask;
+            Vector128<byte> localTwoBytesStringMaskLo = s_base64TwoBytesStringMaskLo;
+
+            for (; i <= length - stride; i += stride)
+            {
+                Vector128<byte> inputVector = Sse2.LoadVector128(inData + i);
+                inputVector = Ssse3.Shuffle(inputVector, localShuffleMask);
+
+                // t0      = [0000cccc|cc000000|aaaaaa00|00000000]
+                Vector128<byte> t0 = Sse2.And(inputVector, tt0);
+                // t1      = [00000000|00cccccc|00000000|00aaaaaa]
+                Vector128<byte> t1 = Sse2.MultiplyHigh(t0.AsUInt16(), tt1).AsByte();
+                // t2      = [00000000|00dddddd|000000bb|bbbb0000]
+                Vector128<byte> t2 = Sse2.And(inputVector, tt2);
+                // t3      = [00dddddd|00000000|00bbbbbb|00000000]
+                Vector128<byte> t3 = Sse2.MultiplyLow(t2.AsUInt16(), tt3).AsByte();
+                // indices = [00dddddd|00cccccc|00bbbbbb|00aaaaaa] = t1 | t3
```

Comment:
nit: the paper looks to differentiate between `BB` and `bbbb` as well as `CC` and `cccc`

Reply:
I am not sure I follow, the comments are 100% copied from https://github.com/WojciechMula/base64simd/blob/master/encode/encode.sse.cpp#L20-L59 🙂

Reply:
> I am not sure I follow, the comments are 100% copied from 

Aren't we then missing the appropriate 3rd-party notice information, copying the relevant licensing information into this file, etc.?
cc: @richlander

Reply:
updated the THIRD-PARTY-NOTICES.txt

Reply:
> I am not sure I follow, the comments are 100% copied from 

I was commenting that the paper, in some places, differentiates between uppercase `BB` and lowercase `bbbb`. The source code doesn't seem to have maintained that in all the various places.

For example, the paper calls out:
> Input to this step are 32-bit words, each having following layout:
> ```
> [bbbbcccc|CCdddddd|aaaaaaBB|bbbbcccc]
> ```
> where bits aaaaaa, BBbbbb, ccccCC and dddddd are 6-bit indices. The output of this step has to be:
> ```
> [00dddddd|00ccccCC|00BBbbbb|00aaaaaa]
> ```

This looks to be particularly important for tracking which bits flow where

Reply:
It would still be nice to see this addressed.